### PR TITLE
James lowenthal/134/replace dropdown with choices js

### PR DESCRIPTION
**Pull request by JamesLowenthal**
Solves https://github.com/ParkingReformNetwork/parking-lot-map/issues/134

Screenshot 2024-02-10 at 2 56 57 PM
Screenshot 2024-02-10 at 2 57 49 PM
Screenshot 2024-02-10 at 2 57 56 PM
Screenshot 2024-02-10 at 2 58 04 PM
Screenshot 2024-02-10 at 2 58 12 PM

### CODE REVIEW

**Comment by Eric-Arellano** [comment]
Thanks! Looks like there's a merge conflict. Also the test upgrade was causing CI to fail; I fixed this in https://github.com/ParkingReformNetwork/parking-lot-map/pull/137.

**Comment by JamesLowenthal** [comment]
> Thanks! Looks like there's a merge conflict. Also the test upgrade was causing CI to fail; I fixed this in #137.

Ah sorry about that, still getting used to the forking workflow.  Also I missed in the readme where it says `npx playwright install`

File: src/css/_header.scss [hunk 1]
```diff
@@ -39,14 +41,38 @@ header {
   .site-title {
     flex: 1;
   }
+
+  #city-dropdown {
+    display: flex;
+    align-items: center;
+    gap: 10px;
```
Comment (Eric-Arellano) [inline src/css/_header.scss:48]: Nit: we've been using `em` rather than `px`. Web dev is not my specialty, but my understanding is that `em` works better for responsive design and things like retina displays. https://www.reddit.com/r/webdev/comments/wlnt6e/should_i_use_emrem_instead_of_px/

File: src/js/setUpSite.js [hunk 2]
```diff
@@ -138,7 +124,8 @@ const loadParkingLot = async (cityId, parkingLayer) => {
     .getLayers()
     .find((city) => city.feature.properties.id === cityId);
   if (!alreadyLoaded) {
-    parkingLayer.addData(await parkingLots[`${cityId}.geojson`]());
+    const lots = await parkingLots;
+    parkingLayer.addData(await lots[`${cityId}.geojson`]());
```
Comment (Eric-Arellano) [inline src/js/setUpSite.js:128]: Why this change? Btw the dynamic import is so that we lazily load the GeoJSON file because they are large and loading everything eagerly slows down the site too much. Implemented recently in https://github.com/ParkingReformNetwork/parking-lot-map/pull/142.

**Review by Eric-Arellano — COMMENTED** [review]
This is great! 🚀 It unblocks us from adding new cities.

File: src/js/setUpSite.js (hunk 2 above)
Comment (JamesLowenthal) [reply src/js/setUpSite.js:128]: I'm reverting it.  I thought it solved an issue I was having with an error I was getting in the console but I can't seem to reproduce the error.  I'm finding it a bit difficult to make sure the version of the app I have served is up to date with my changes and I've gotten what I think is a corrupted cache a number of times (weird segfauls/mutex lock complaints?).  Anyway sorry that crept in.

File: src/css/_header.scss (hunk 1 above)
Comment (JamesLowenthal) [reply src/css/_header.scss:48]: sgtm, thanks!

File: src/js/setUpSite.js (hunk 2 above)
Comment (Eric-Arellano) [reply src/js/setUpSite.js:128]: Hm, possibly Parcel bugs. Sometimes it can help to `rm -rf .parcel-cache`.

File: src/css/_header.scss (hunk 1 above)
Comment (JamesLowenthal) [reply src/css/_header.scss:48]: Honestly I'm finding it difficult to redo my styling using em without breaking the layout.  I've changed for the gap and media query since that seemed to work fine.  I'll confess my CSS workflow (as well as many of the webdevs I know) is basically hack on it until it does what I want but I won't tell you I understand it particular well.  This url that was in the thread you sent seems pretty good for understanding it better: https://www.joshwcomeau.com/css/surprising-truth-about-pixels-and-accessibility/. 

It looks like in most cases you'd switch back and forth anyway.  That all being said I'd say most of the frontend codebases I seen use a mix of the two and px is still used in many places I don't think many web-devs would agree with "never uses px" but maybe I'm wrong.

If it's obvious to you how I can redo this layout using em I'm happy to make the changes (it doesn't seem to be as simple as calculating em and subbing those values in)

File: src/js/setUpSite.js (hunk 2 above)
Comment (JamesLowenthal) [reply src/js/setUpSite.js:128]: Yeah that's what I ended up doing.

File: src/css/_header.scss (hunk 1 above)
Comment (Eric-Arellano) [reply src/css/_header.scss]: That's fine with me. Thanks, James!

**Review by Eric-Arellano — APPROVED** [review]
Huzzah, thanks! 

Btw with at least the open source repos I've worked on, the PR title conventionally wouldn't include your name. It sometimes includes the issue number, but usually, that's only in the body. 

Here, it can be something like "Switch dropdown component to choices.js"

See https://www.pantsbuild.org/2.18/docs/contributions#opening-a-pull-request for how one project I was a previous maintainer for does PR titles. Of course, other projects may vary.
